fix: 🐛 Updated levelup version to 4.2.0 fixing audit

srepollock opened the pull request:
Closes: #20

srepollock committed:
fix: 🐛 Updated levelup version to 4.2.0 fixing audit

Closes: #20

favna commented:
@mafintosh please merge this and create an updated release

akoushke commented:
I would like to request for a change, if that's okay. There is another vulnerability in `levelup`. the problem had been fixed already form their side https://github.com/Level/levelup/issues/676. However, I'm not sure if it will be a patch/minor version bump. if not, then we need to update this package.json as well.
refer to this picture for the vulnerabilities that I had faced using `rollup` plugin.

<img width="512" alt=" " src="https://user-images.githubusercontent.com/14226615/65379059-19565380-dc90-11e9-8b83-d345d3ce433a.png">

vweevers commented:
@srepollock This PR is incomplete. There have been breaking changes between levelup 0.x and 4 you'll need to account for.

srepollock committed:
fix(index.js): updated to comply with v4 of levelup

srepollock commented on index.js:
```diff
@@ -1,6 +1,7 @@
 var fs = require('level-filesystem');
 
-var db = levelup('level-filesystem', {db:leveljs});
+var db = levelup(leveldown('level-filesystem'), {db:leveljs});
```
@vweevers I've updated to use the leveldown system. I ran with `node index.js example.js` and it created a new file with some logs in the main project directory. They had `.log` files and I don't know the exact contents as they were written in buffered data. I believe this may be a fix, but request a review

Vehmloewff commented:
@mafintosh, have you had a chance to look at this yet?  It would help us out a lot if this were fixed.

mafintosh commented:
I don’t have any time to review this. If someone wants to take it over let me know.

srepollock commented:
@mafintosh user @lcsvcn has approved this and I addressed the issues @Vehmloewff commented on. Is there any more reason stopping this from being merged in?

srepollock commented:
I apologize and would like to update this thread for the sake of @trevorblades . I have accidentally deleted the GitHub repo referenced in the PR here. I have [recreated the repo and made the necessary changes here](https://github.com/srepollock/browserify-fs). I will have a [release set up](https://www.npmjs.com/package/srepollock-browserify-fs) as well to pull from NPM, but I would suggest that this PR get some movement.

If because I deleted the repo there is no way to merge in this PR, please @ me here and I can submit a new one with the new repo. Cheers

favna commented:
@srepollock github maintains the code changes of deleted repos, thankfully. In the past it would've auto closed this PR but no more.

trevorblades commented:
Thanks for restoring that repo @srepollock!

jdalrymple commented:
Any updates?

SamsonChoo commented:
Could someone teach me how to use @srepollock's fixed version of the repo instead of this version? Seeing that this PR will not be merged anytime soon.

srepollock commented:
@SamsonChoo `npm i srepollock-browserify-fs`
https://www.npmjs.com/package/srepollock-browserify-fs

penfold45 commented:
Hi this appears to still be an open issue. Is there a way to move this along and get it resolved? Happy to help in anyway I can

srepollock commented:
@penfold45 currently the solution is to fork and fix yourself or to use someone else's fix. The developer seems to be inactive on here at this time.
